connector/dialect.tdd

```diff
@@ -379,650 +183,568 @@
-      <formula>(%1 &lt;&gt; %2)</formula>
-      <argument type='str' />
+    <function group='logical' name='NULLIF' return-type='int'>
+      <formula>NULLIF(%1)</formula>
+      <argument type='int' />
+    </function>
+    <function group='logical' name='NULLIF' return-type='str'>
+      <formula>NULLIF(%1)</formula>
       <argument type='str' />
     </function>
-    <function group='operator' name='!=' return-type='bool'>
-      <formula>(%1 &lt;&gt; %2)</formula>
-      <argument type='datetime' />
+    <function group='logical' name='NULLIF' return-type='datetime'>
+      <formula>NULLIF(%1)</formula>
       <argument type='datetime' />
     </function>
-    <function group='operator' name='%' return-type='real'>
-      <formula>(CASE WHEN %2 = 0 THEN NULL ELSE %1 - FLOOR(ABS(%1)/ABS(%2)) * ABS(%2) * SIGN(%1) END)</formula>
+    <function group='numeric' name='DIV' return-type='int'>
+      <formula>%1 / %2</formula>
+      <argument type='int' />
+      <argument type='int' />
+    </function>
+    <function group='numeric' name='DIV' return-type='real'>
+      <formula>%1 / %2</formula>
       <argument type='real' />
       <argument type='real' />
     </function>
-    <function group='operator' name='%' return-type='int'>
-      <formula>(%1 % %2)</formula>
-      <argument type='int' />
+    <function group='numeric' name='DIV' return-type='real'>
+      <formula>%1 / %2</formula>
       <argument type='int' />
+      <argument type='real' />
     </function>
-    <function group='operator' name='&amp;&amp;' return-type='bool'>
-      <formula>(%1 AND %2)</formula>
-      <argument type='bool' />
-      <argument type='bool' />
+    <function group='numeric' name='DIV' return-type='real'>
+      <formula>%1 / %2</formula>
+      <argument type='real' />
+      <argument type='int' />
     </function>
     <function group='operator' name='*' return-type='real'>
-      <formula>(%1 * %2)</formula>
+      <formula>%1 * %2</formula>
       <argument type='real' />
       <argument type='real' />
     </function>
     <function group='operator' name='*' return-type='int'>
-      <formula>(%1 * %2)</formula>
+      <formula>%1 * %2</formula>
+      <argument type='int' />
+      <argument type='int' />
+    </function>
+    <function group='operator' name='*' return-type='real'>
+      <formula>%1 * %2</formula>
+      <argument type='int' />
+      <argument type='real' />
+    </function>
+    <function group='operator' name='*' return-type='real'>
+      <formula>%1 * %2</formula>
+      <argument type='real' />
       <argument type='int' />
+    </function>
+    <function group='operator' name='+' return-type='int'>
+      <formula>+%1</formula>
       <argument type='int' />
     </function>
     <function group='operator' name='+' return-type='real'>
-      <formula>(%1 + %2)</formula>
+      <formula>+%1</formula>
+      <argument type='real' />
+    </function>
+    <function group='operator' name='+' return-type='real'>
```

Comment:
Did you try a mix of arithmetic operators in the same query? Does Tableau add () around them? 
If not we need to add () them in the formulas, because _A + B x C_ does not equal _(A + B) x C_ and this could be a problem.

Reply:
Decided in standup today to re-add parentheses to the formulas.  I didn't test the mix of operators previously.  
We also decided we will run TDVT in the future when it is setup to check for correctness of the dialect file.